src/middlewared/middlewared/plugins/disk.py

```diff
@@ -60,15 +61,115 @@ def get_name(self, disk):
         Helper method to get all disks that are not in use, either by the boot
         pool or the user pools.
         """
-        disks = await self.query([('name', 'nin', await self.__get_reserved())])
+        disks = await self.query([('name', 'nin', await self.get_reserved())])
 
         if join_partitions:
             for disk in disks:
                 disk["partitions"] = await self.__get_partitions(disk)
 
         return disks
 
-    async def __get_reserved(self):
+    @accepts(Dict(
+        'options',
+        Bool('unused', default=False),
+    ))
+    def get_encrypted(self, options):
+        """
+        Get all geli providers
+
+        It might be an entire disk or a partition of type freebsd-zfs
+        """
+        providers = []
+
+        disks_blacklist = []
+        if options['unused']:
+            disks_blacklist += self.middleware.call_sync('disk.get_reserved')
+
+        geom.scan()
+        klass_part = geom.class_by_name('PART')
+        klass_label = geom.class_by_name('LABEL')
+        if not klass_part:
+            return providers
+
+        for g in klass_part.geoms:
+            for p in g.providers:
+
+                if p.config['type'] != 'freebsd-zfs':
+                    continue
+
+                disk = p.geom.consumer.provider.name
+                if disk in disks_blacklist:
+                    continue
+
+                try:
+                    subprocess.run(
+                        ['geli', 'dump', p.name],
+                        stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, check=True,
+                    )
+                except subprocess.CalledProcessError:
+                    continue
+
+                dev = None
+                if klass_label:
+                    for g in klass_label.geoms:
+                        if g.name == p.name:
+                            dev = g.provider.name
+                            break
+
+                if dev is None:
+                    dev = p.name
+
+                providers.append({
+                    'name': p.name,
+                    'dev': dev,
+                    'disk': disk
+                })
+
+        return providers
+
+    @accepts(
+        List('devices', items=[Str('device')]),
+        Str('passphrase'),
```

Comment:
I've just merged https://github.com/freenas/freenas/pull/1017 now we can use this to mask `passphrase` in `core.get_jobs`